lib/routeLoader.js

```diff
@@ -0,0 +1,94 @@
+const fs = require('fs');
+const glob = require('glob');
+const path = require('path');
+const _ = require('lodash');
+const async = require('async');
+const defaults = require('./defaults');
+const getCompletePath = require('./getCompletePath');
+
+module.exports = (server, options, next) => {
+  const load = (loadOptions, loadDone) => {
+    const stub = () => {};
+    loadDone = loadDone || stub;
+    const settings = _.clone(loadOptions);
+    _.defaults(settings, defaults);
+    // the main flow is here:
+    async.autoInject({
+      // confirm that settings.path exists and is a directory:
+      confirmDirectoryExists(done) {
+        fs.exists(settings.path, (exists) => {
+          if (!exists) {
+            server.log(['hapi-route-loader', 'warning'], { message: 'path doesnt exist', path: settings.path });
+            return done();
+          }
+          fs.stat(settings.path, (err, stat) => {
+            if (err) {
+              return done(err);
+            }
+            if (!stat.isDirectory()) {
+              server.log(['hapi-route-loader', 'warning'], { message: 'path not a directory', path: settings.path });
+              return done(`path ${settings.path} not a directory`);
+            }
+            return done();
+          });
+        });
+      },
+      // get the list of all matching route-containing files
+      files(confirmDirectoryExists, done) {
+        glob('**/*.js', {
+          cwd: settings.path
+        }, (globErr, files) => {
+          if (globErr) {
+            return done(globErr);
+          }
+          done(null, files);
+        });
+      },
+      // for each filename, get a list of configured routes defined by it
+      configureAllRoutes(files, done) {
+        const routeConfigs = {};
+        files.forEach((fileName) => {
```

Comment:
lets create a separate function for getting routes from a file and another getting a route from an export and add tests for those functions individually